src/main/java/spoon/template/TemplateMatcher.java

```diff
@@ -62,12 +62,9 @@
 public class TemplateMatcher implements Filter<CtElement> {
 
 	/**
-	 * Collects all AST nodes, which has to be substituted, because they represents a template parameter declared by {@link TemplateParameter}
+	 * Searches for all invocations of {@link TemplateParameter#S()} in "root", a CtClass model of {@link Template}
 	 *
-	 * Searches for all invocations of {@link TemplateParameter#S()} in CtClass model of {@link Template}
 	 * @param root CtClass model of {@link Template}
-	 * @return List of all variables - only these parameters which are declared by {@link TemplateParameter}.
-	 * It doesn't includes references to parameters annotated by {@link Parameter}
 	 */
```

Comment:
I don't understand. does it return a list of variables or parameters? or a list of invocations?